### PR TITLE
Implement Basic ManageBac Overview and CAS screens

### PR DESCRIPTION
**Pull request by FoxInFlame**
This pull request will add screens for Overview and CAS inside the Managebac tab of Sardonyx.
Some additional screens to accommodate those two screens may be added, such as detail screens for each CAS experience.

More sophistication will be brought to these screens in later pull requests.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,6 +11,7 @@
 /node_modules
 
 # misc
+.env
```
Comment (natsuozawa) [inline .gitignore]: It doesn't look like .env and .env.production are correctly ignored...

File: src/components/CalendarDate.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    'Dec'
+  ];
+
+  render() {
+    return (
+      <View style={calendarStyles.container}>
+        <View style={calendarStyles.calendar}>
+          <View style={calendarStyles.month}>
+            <Text style={calendarStyles.monthText}>
+              {this.shortMonthNames[this.props.date.getMonth()]}
```
Comment (natsuozawa) [inline src/components/CalendarDate.js]: You may want to use 

```js
new Intl.DateTimeFormat('en-US', { month: 'short'}).format(this.props.date);
```

File: src/components/ExpandableCard.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    borderColor: colors.gray1,
+    borderWidth: 1,
+    borderRadius: 4,
+    overflow: 'hidden'
+  },
+  hidden: {
+    height: 0
+  },
+  shown: {
+
```
Comment (natsuozawa) [inline src/components/ExpandableCard.js]: Remove empty properties?

File: src/components/CalendarDate.js (hunk 2 above)
Comment (natsuozawa) [inline src/components/CalendarDate.js:33]: You may want to use 

```js
new Intl.DateTimeFormat('en-US', { month: 'short'}).format(this.props.date);
```

File: .gitignore (hunk 1 above)
Comment (natsuozawa) [inline .gitignore]: .env is not correctly ignored?

File: src/components/ExtendedDeadline.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+          method: 'GET',
+          headers: {
+            'Login-Token': credentials
+          },
+          mode: 'no-cors'
+        }).then(response => {
+          if (response.status === 200) {
+            resolve(JSON.parse(response.headers.map['managebac-data']));
+          }
+        });
```
Comment (natsuozawa) [inline src/components/ExtendedDeadline.js]: Error handling please 
(Promise rejection)

File: src/components/UpcomingExpandableCard.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+import { colors } from '../styles';
+
+export default class UpcomingExpandableCard extends ExpandableCard {
+  constructor(props) {
+    super(props);
+    this.state = {
+      expanded: true
+    };
+    this._getGroupClassName = this._getGroupClassName.bind(this);
+    this._navigateToUpcomingEventScreen = this._navigateToUpcomingEventScreen.bind(
```
Comment (natsuozawa) [inline src/components/UpcomingExpandableCard.js]: One line is fine here 
```js
this._navigateToUpcomingEventScreen = this._navigateToUpcomingEventScreen.bind(this);
```

File: src/styles.js [hunk 6]
```diff
@@ -101,8 +117,68 @@ const styles = StyleSheet.create({
...
+      color = '#1aaf5d';
+      break;
+    case 'Extended Essay':
+      color = '#cc3333';
+      break;
+    default:
+      return;
+  }
+  return color;
+};
```
Comment (natsuozawa) [inline src/styles.js:182]: This is unsustainable. Let's parse the color server side and pass it along with the other label data.

File: src/screens/LoginScreen.js [hunk 7]
```diff
@@ -190,7 +192,7 @@ class LoginForm extends React.Component {
           })}
```
Comment (natsuozawa) [inline src/screens/LoginScreen.js:192]: The styling seems a bit broken here and the checkbox is ugly (the font doesn't match, etc)

I don't think we need a checkbox, just include a statement that says 
"By signing in, you agree to our Terms of Service and Privacy Policy"

Link to `BASE_URL + '/terms'` and `BASE_URL + '/privacy'`

File: src/components/ExtendedDeadline.js [hunk 8]
```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+
+import {
+  View,
+  Text,
+  ActivityIndicator,
+  StyleSheet,
+  FlatList
```
Comment (natsuozawa) [inline src/components/ExtendedDeadline.js]: FlatList is not used

**Review by natsuozawa — CHANGES_REQUESTED** [review]
Please make sure the app is stable before you merge

**Review by natsuozawa — APPROVED** [review]

File: src/screens/LoginScreen.js (hunk 7 above)
Comment (natsuozawa) [reply src/screens/LoginScreen.js:192]: We will work on this later.

Comment (FoxInFlame) [reply src/screens/LoginScreen.js:192]: Understood.

File: src/styles.js (hunk 6 above)
Comment (FoxInFlame) [reply src/styles.js:182]: Understood.

File: src/components/UpcomingExpandableCard.js (hunk 5 above)
Comment (FoxInFlame) [reply src/components/UpcomingExpandableCard.js]: Fixed.

**Pull request merged**